### PR TITLE
Add nested pagination docs

### PR DESCRIPTION
**Pull request by KenitoInc**
Add sample code on how we can handle nested pagination in OData Client

*Expected sample output*
```
Person Page 1:
        Person Name: Russell
            Trips Page 1:   
                TripID: 0 - Name: $Trip in US
                TripID: 1003 - Name: $Trip in Beijing

            Trips Page 2:
                TripID: 1007 - Name: $Honeymoon

        Person Name: Scott
            Trips Page 1:
                TripID: 0 - Name: $Trip in US
                TripID: 2004 - Name: $Trip in Beijing
         .............................................
        .............................................
        .............................................

Person Page 2:
        Person Name: Marshall
            Trips Page 1:
                .............................................
                ............................................

        Person Name: Ryan
            Trips Page 1:
               .............................................
               .............................................

            Trips Page 2:
               .............................................
               .............................................
```

### CODE REVIEW

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 1]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
...
+
+    // Get the next link, and continue while there is a next link.
+    while ((nextLink = response.GetContinuation()) != null);
+}
+catch (DataServiceQueryException ex)
+{
+    throw new ApplicationException(
+        "An error occurred during query execution.", ex);
+}
+```
```
Comment (xuzhg) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md]: what's the result looks like?

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 2]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
...
+                    t.TripId, t.Name);
+            }
+
+            while (nextTripsLink != null)
+            {
+                Console.WriteLine("\t\t******Load Trips******");
+
+                // Load the next page of Trips.
+                var tripsResponse = context.LoadProperty(p, "Trips", nextTripsLink);
+                nextTripsLink = tripsResponse.GetContinuation();
```
Comment (odero) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md]: This line fails unless you have it as (I used .NetCore)
```c#
nextTripsLink = tripsResponse.GetContinuation(p.Trips);
```

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 3]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
...
+DataServiceQueryContinuation<Person> nextLink = null;
+int pageCount = 0;
+int innerPageCount = 0;
+
+try
+{
+    // Execute the query for all people and related trips,
+    // and get the response object.
+    var response =
+        context.People.AddQueryOption("$expand", "Trips")
```
Comment (odero) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md]:
```suggestion
        context.People.Expand("Trips")
```
Seems more straightforward

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 4]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
...
+                response.GetContinuation(p.Trips);
+
+            foreach (Trip t in p.Trips)
+            {
+                // Print out the trips in the first page.
+                Console.WriteLine("\t\tTripID: {0} - Name: ${1}",
+                    t.TripId, t.Name);
+            }
+
+            while (nextTripsLink != null)
```
Comment (odero) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md]: Cautionary FYI: For this particular example using TripPin, the body of this while loop seems to never be executed since the number of nested results (i.e. trips) is always smaller than the page size

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 5]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
     response = context.Execute<Person>(token);
 }
 ```
+
+## Nested Pagination
+There are usecases where we need to load pages of entities and also pages of their related entities.
+This example returns related Trips for each Person entity and uses a do�while loop to load Person entities pages and a nested while loop to load pages of related Trips from the data service.
```
[the remark this answers is not in the document]
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 6]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
...
+
+``` csharp
+var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
+DataServiceQueryContinuation<Person> nextLink = null;
+int pageCount = 0;
+int innerPageCount = 0;
```
[the remark this answers is not in the document]
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 7]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
     response = context.Execute<Person>(token);
 }
 ```
+
+## Nested Pagination
+There are usecases where we need to load pages of entities and also pages of their related entities.
```
[the remark this answers is not in the document]
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 8]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
...
+This example returns related Trips for each Person entity and uses a do�while loop to load Person entities pages and a nested while loop to load pages of related Trips from the data service.
+
+``` csharp
+var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
+DataServiceQueryContinuation<Person> nextLink = null;
```
[the remark this answers is not in the document]
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 9]
```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
...
+            // Load the new page from the next link URI.
+            response = context.Execute<Person>(nextLink)
+                as QueryOperationResponse<Person>;
+        }
+
+        // Enumerate the People in the response.
+        foreach (Person p in response)
+        {
+            Console.WriteLine("\Person Name: {0}", p.Name);
+            Console.WriteLine("\Trips Page {0}:", ++innerPageCount);
```
[the remark this answers is not in the document]
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 2 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: The problem was this line 
`DataServiceQueryContinuation nextTripsLink = response.GetContinuation(p.Trips);`
#Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 3 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 4 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: In do...while, the code block is executed atleast once.

Comment (odero) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: I'm referring to the `while (nextTripsLink != null)` loop. I placed a break point inside and it never gets executed even once

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 1 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 4 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md:85]: We don't have server side pagination in trippin service. The sample code is just a guide for now.

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 10]
```diff
@@ -19,15 +19,127 @@ Example:
...
+    do
+    {
+        // Write the page number.
+        Console.WriteLine("Person Page {0}:", ++pageCount);
+
+        // If nextLink is not null, then there is a new page to load.
+        if (nextLink != null)
+        {
+            // Load the new page from the next link URI.
+            response = await context.ExecuteAsync<Person>(nextLink)
```
Comment (gathogojr) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md:65]: @KenitoInc I'm comparing this line with this [other one](https://github.com/MicrosoftDocs/OData-docs/pull/114/files#diff-f9bf5ec4bf534abf08213e96615c00dbR31) - Line 31 - and wondering why the latter uses `context.Execute<Person>(nextLink)` and not `await context.ExecuteAsync<Person>(nextLink)` like it is here

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 11]
```diff
@@ -19,15 +19,127 @@ Example:
...
+                as QueryOperationResponse<Person>;
+        }
+
+        // Enumerate the Person(s) in the response.
+        foreach (Person p in response)
+        {
+            Console.WriteLine("\tPerson Name: {0}", p.FirstName);
+            Console.WriteLine("\tTrips Page {0}:", ++innerPageCount);
+            // Get the next link for the collection of related Trips.
+            DataServiceQueryContinuation nextTripsLink =
```
Comment (gathogojr) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md:75]:
```suggestion
            DataServiceQueryContinuation tripsNextLink =
```

**Review by gathogojr — APPROVED** [review]
Left some 2 comments, otherwise very helpful documentation

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 12]
```diff
@@ -19,15 +19,127 @@ Example:
...
 // Get the first page
 var response = await context.People.ExecuteAsync() as QueryOperationResponse<Person>;
 
 // Loop if there is a next link
-while ((token = response.GetContinuation()) != null)
+while ((nextLink = response.GetContinuation()) != null)
 {
     // Get the next page
-    response = await context.ExecuteAsync<Person>(token) as QueryOperationResponse<Person>;
+    response = context.Execute<Person>(nextLink);
```
Comment (habbes) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md:31]: Why the change from `ExecuteAsync()` to `Execute()`?

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 13]
```diff
@@ -19,15 +19,127 @@ Example:
...
+                TripID: 0 - Name: $Trip in US
+                TripID: 1003 - Name: $Trip in Beijing
+
+        Trips Page 2:
+                TripID: 1007 - Name: $Honeymoon
+
+        Person Name: Scott
+        Trips Page 2:
+                TripID: 0 - Name: $Trip in US
+                TripID: 2004 - Name: $Trip in Beijing
```
Comment (habbes) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md:126]: Why does Scott's trips start at page 2?

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 14]
```diff
@@ -19,15 +19,127 @@ Example:
...
+
+## Nested Pagination
+There are instances where we need to load pages of entities as well as pages of their related entities.
+The example below returns related Trips for each Person entity from the data service. It uses a do...while loop to paginate through Person entities and a nested while loop to paginate through the related Trips.
+We need to enumerate the response before calling `GetContinuation` on it.
+
+``` csharp
+var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
+var pageCount = 0;
+var innerPageCount = 0;
```
Comment (habbes) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md:43]: Shouldn't the `innerPageCount` variable be initialized at 0 for each person?

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 13 above)
Comment (habbes) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md:126]: I think it's cause both persons' loops share and update the same `innerPageCount` variable without re-initializing it to 0.

File: Odata-docs/client/v6/deal-with-server-driven-paging.md [hunk 15]
```diff
@@ -19,15 +19,127 @@ Example:
...
+        Trips Page 3:
+               .............................................
+               ............................................
+
+        Person Name: Ryan
+        Trips Page 4:
+               .............................................
+               .............................................
+
+        Trips Page 5:
```
Comment (habbes) [inline Odata-docs/client/v6/deal-with-server-driven-paging.md:142]: I assume each person's trips page should start at Page 1. Is this "global" page counting intentional.

Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 14 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 13 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved
Every person's trips start from 1

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 12 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 11 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md]: #Resolved

File: Odata-docs/client/v6/deal-with-server-driven-paging.md (hunk 10 above)
Comment (KenitoInc) [reply Odata-docs/client/v6/deal-with-server-driven-paging.md:72]: #Resolved
